satpy/readers/viirs_sdr.py

```diff
@@ -107,8 +107,10 @@ def _get_invalid_info(granule_data):
                 'SVM14': 'VIIRS-M14-SDR',
                 'SVM15': 'VIIRS-M15-SDR',
                 'SVM16': 'VIIRS-M16-SDR',
-                'IVCDB': 'VIIRS-DualGain-Cal-IP'
-                }
+                'IVCDB': 'VIIRS-DualGain-Cal-IP',
+                'SATMS': 'ATMS-SDR',
+                'GATMO': 'ATMS-SDR-GEO',
+                'TATMS': 'ATMS-TDR'}
```

Comment:
Can we maybe make this `DATASET_KEYS` part of the class object, then refactor the class to be more of a base class and have a VIIRS subclass and an ATMS subclass?

Reply:
I have made a JPSS SDR base class as suggested but still keeping the `DATASET_KEYS` as a common variable outside the class. Should I include it in the base class?

I tried adding some test coverage. Not sure about this probably rather old (> 7 years) try-except:

https://github.com/pytroll/satpy/blob/11708d0d4cfd39b9eb62e4177ca3eb4a63c49653/satpy/readers/viirs_sdr.py#L151

If that is ever being used? But it was not tested. It is now, but I guess this line is from the time before Dask? Therefore the change here.